### ISSUE TITLE
Programatically updating Operation Summary / Description

### ISSUE THREAD

**Issue by yoandinkov** [domaindrivendev/Swashbuckle.AspNetCore#2199]
Hi all,

I have a requirement to programmatically update the text of operation's summary (or) description.

So far, I have found from README documentation, that I can access the operation data, via IOperationFilter.

I've tried following approach (with properly registering the Filter in `SwaggerGen`
```csharp
public class TestOperationFilter : IOperationFilter
{
      public void Apply(OpenApiOperation operation, OperationFilterContext context)
      {
          operation.Summary = $"{operation.Summary} Test";
      }
}
```

with no avail. It looks like that `operation.Summary` is with value `null` even though I am setting it up via `[SwaggerOperation(Summary="...")]` attribute property. Also even "hardcoding" the "Test" prefix is not visible in the generated Open Api file (swagger.json).

The second more hacky approach was to somehow update/set value of the property via `ActionModelConvention` (again registering it correcly in `AddMvc` conventions)

With the following code.
```csharp
public class TestConvention : IActionModelConvention
{
    public void Apply(ActionModel action)
    {
        var test = action.Attributes.OfType<SwaggerOperationAttribute>().First();
        var idx = action.Attributes.IndexOf(test);
        ((SwaggerOperationAttribute) action.Attributes[idx]).Summary = "Test";
    }
}
```

Here I've noticed that `action.Attributes` is of type `IReadOnlyList<object>` which leaves me to believe that these values should not be updateable (hence the weird/hacky index location).

Yet, none of these methods works to dynamically update summary (and in the same fashion) description of the operations.

I am using latest stable Swashbuckle nuget package with .net core 3.1

**Comment by domaindrivendev** [comment]
You can add summaries declaratively via “SwaggerOperationAttribute” or imperatively via a registered IOperationFilter. NOTE: for the former you need to explicitly enable SwaggerAnnotations by installing the additional package and invoking “EnableAnnotations” in the Swagger config as described in the readme.

Both of these approaches are tested as part of the build and should 100% work so there must be something else going on if they’re not.

To help troubleshoot, please create a minimal app (ie starting from empty project) that repro’s your issue and post to github so I can pull down and dig deeper.

**Comment by yoandinkov** [comment]
@domaindrivendev Thanks for the quick response and the focus on the `EnableAnnotations` method. My problem was timing (or code location) in this case. I was first registering the `IOperationFilter` and afterwards (several lines down the config) was calling `EnableAnnotations`. After I've fixed the order of method calls/registering all worked as supposed.

Maybe the next step here will be to add some indication of this in the documentation, otherwise, I am closing the issue as my problem was fixed and my question was answered appropriately.
